workers/unity/Assets/Playground/Scripts/EntityInitialization/GameObjectInitializationSystem.cs

```diff
@@ -12,25 +12,32 @@ namespace Playground
     /// <summary>
     ///     Creates a companion gameobject for newly spawned entities according to a prefab definition.
     /// </summary>
-    [UpdateInGroup(typeof(SpatialOSReceiveGroup.EntityInitialisationGroup))]
+    [UpdateBefore(typeof(SpatialOSReceiveGroup.GameObjectReceiveGroup))]
+    [UpdateAfter(typeof(SpatialOSReceiveGroup.InternalSpatialOSReceiveGroup))]
     internal class GameObjectInitializationSystem : ComponentSystem
     {
+        public struct EntityGameObjectHandle : ISystemStateComponentData
+        {
+        }
+
         private struct AddedEntitiesData
         {
             public readonly int Length;
             public EntityArray Entities;
             [ReadOnly] public ComponentDataArray<Prefab.Component> PrefabNames;
             [ReadOnly] public ComponentDataArray<Transform.Component> Transforms;
             [ReadOnly] public ComponentDataArray<SpatialEntityId> SpatialEntityIds;
-            [ReadOnly] public ComponentDataArray<NewlyAddedSpatialOSEntity> NewlyCreatedEntities;
+            [ReadOnly] public SubtractiveComponent<EntityGameObjectHandle> NewlyCreatedEntities;
         }
 
         private struct RemovedEntitiesData
         {
             public readonly int Length;
             public EntityArray Entities;
-            [ReadOnly] public ComponentDataArray<GameObjectReferenceHandle> GameObjectReferenceHandles;
-            public SubtractiveComponent<GameObjectReference> NoGameObjectReference;
+            [ReadOnly] public SubtractiveComponent<Prefab.Component> PrefabNames;
+            [ReadOnly] public SubtractiveComponent<Transform.Component> Transforms;
+            [ReadOnly] public SubtractiveComponent<SpatialEntityId> SpatialEntityIds;
+            [ReadOnly] public ComponentDataArray<EntityGameObjectHandle> NewlyCreatedEntities;
```

Comment:
General note - any `SubtractiveComponent<T>` should be called something like `No T Here` or similar.